### ISSUE TITLE
Add code coverage uploads for non-amd64 architectures

### ISSUE THREAD

**Issue by codefromthecrypt** [tetratelabs/wazero#598]
**Is your feature request related to a problem? Please describe.**
https://app.codecov.io/gh/tetratelabs/wazero doesn't include all main sources, rather just amd64

We should have all sources. There are notably more for arm64, but also a few for neither (risc-v)

**Describe the solution you'd like**
I'd like codecov to present a merged view of all sources, which are materialized by emulated arm64 and risc-v also uploading reports.

**Describe alternatives you've considered**
not doing anything

**Additional context**
https://github.com/tetratelabs/wazero/pull/596#issuecomment-1140124291

**Comment by codefromthecrypt** [comment]
note: the GitHub action for codecov doesn't yet support non-amd64 https://github.com/codecov/uploader/issues/523

**Comment by codefromthecrypt** [comment]
TL;DR; let's use the normal bash script until non-amd64 works on the action, or they drop the bash script

if they really delete [the bash script](https://codecov.io/bash) before fixing non-amd64 a complicated workaround would be to have a fan-in upload stage. basically run all the coverage in a matrix and upload artifacts with the coverage.txt. block on that, download all the coverage.txt into a different directory or path and then use codecov/codecov-action@v2 action to send them at the same time.

The ideal way out is to wait for non-amd64 to work and use the bash script meanwhile.

**Comment by philippgille** [comment]
The referenced [PR](https://github.com/codecov/uploader/issues/523) for the codecov uploader is now closed since 2 days and they link to arch-specific uploader binaries, so we could give it a try.

One question:

Currently the coverage only runs as part of the [`test_amd64` job](https://github.com/tetratelabs/wazero/blob/9138b6f722ca855c8f498df6749f302a8a4e9ffa/.github/workflows/commit.yaml#L60) [here](https://github.com/tetratelabs/wazero/blob/9138b6f722ca855c8f498df6749f302a8a4e9ffa/.github/workflows/commit.yaml#L92-L100).

Is your plan to
1. Rename that to be arch independent and add other architectures to the [matrix](https://github.com/tetratelabs/wazero/blob/9138b6f722ca855c8f498df6749f302a8a4e9ffa/.github/workflows/commit.yaml#L65)?
2. Or add arm64 and riscv64 jobs next to it?
3. Or add coverage tracking to the [`test_scratch` job](https://github.com/tetratelabs/wazero/blob/9138b6f722ca855c8f498df6749f302a8a4e9ffa/.github/workflows/commit.yaml#L102), which already has all three architectures in [its matrix](https://github.com/tetratelabs/wazero/blob/9138b6f722ca855
c8f498df6749f302a8a4e9ffa/.github/workflows/commit.yaml#L107-L114)?
